include/ceed/ceed.h

```diff
@@ -140,14 +140,20 @@ typedef int32_t CeedInt;
 typedef ptrdiff_t CeedSize;
 
 /// Scalar (floating point) types
-///
 /// @ingroup Ceed
 typedef enum {
   /// Single precision
   CEED_SCALAR_FP32,
   /// Double precision
-  CEED_SCALAR_FP64
+  CEED_SCALAR_FP64,
+  /// Total number of allowed scalar precision types
+  CEED_NUM_PRECISIONS,
 } CeedScalarType;
+/// Struct for holding data in multiple precisions for mixed-precision-enabled
+/// backends
+typedef struct {
+  void *values[CEED_NUM_PRECISIONS]; // Size equals CEED_NUM_PRECISIONS
+} CeedScalarArray;
```

Comment:
This automatically gets the right sizes. Can we move `CeedScalarArray` to `ceed-impl.h` or `backend.h`?